pkg/controller/controller_instance.go

```diff
@@ -329,6 +349,68 @@ func (c *controller) initOrphanMitigationCondition(instance *v1beta1.ServiceInst
 	return false, nil
 }
 
+// setNextProvisionRetryTime adds the instance name to the map recording the next earliest time to
+// retry provisioning
+func (c *controller) setNextProvisionRetryTime(instance *v1beta1.ServiceInstance) {
+	key, err := cache.DeletionHandlingMetaNamespaceKeyFunc(instance)
+	if err != nil {
+		glog.Errorf("Couldn't get key for object %+v: %v", instance, err)
+		return
+	}
+	c.provisionRetryQueue.mutex.Lock()
+	c.provisionRetryQueue.retryTime[key] = time.Now().Add(c.provisionRetryQueue.retryInterval)
```

Comment:
Can you take a look at `NewItemExponentialFailureRateLimiter()` in `github.com/kubernetes-incubator/service-catalog/vendor/k8s.io/client-go/util/workqueue/default_rate_limiters.go` ? 

I think adding a nice exponential backoff instead of a constant delay should be fairly simple. The rate limiter has a `When()` method you can call to get the next time. 

But I haven't really looked at the rate limiters closely. So I may be wrong :)

Reply:
An exponential backoff is ideal, it is what we were supposed to be doing here, but we constantly reset the notion of failure/success when we do status updates or complete the orphan mitigation and we never quit after max retries, it just restarts all over.

To use exponential backoff we need to maintain more state, including the # of retries.

Ideally we would have a queue much like today that has very short backoff/retry for kube reconciliation, but we probably need another queue that is for remote broker operations that has a larger backoff.  I 100% agree that exponential backoff is the right fix, but its going to take a large rework.  The only goal of this PR is to stop us from pounding on the broker.

Reply:
No, I meant just create a new rate limiter instead of the `retryTime` map. The rate limiter keeps track of everything. All you need to do is call `rateLimiter.When(instance)` here and it'll keep track of the individual delays for each ServiceInstance.

Reply:
I see what you are saying Marko, great idea.  I think I'll have to keep the `retryMap` in addition to adding the `rateLimter`.  The `rateLimiter` doesn't have support for checking if an item is in queue and getting the next execution time.  But I really like the idea of exponential backoff.